trace/noop.go

```diff
@@ -21,7 +21,7 @@ type noOpTracer struct {
 }
 
 func (n noOpTracer) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
-	return n.t.Start(ctx, spanName, opts...)
+	return n.t.Start(ctx, spanName, opts...) //nolint:spancheck
```

Comment:
the linter suggests there is a possible memory leak here. Should we end the spam right away for the noop tracer instead of silencing the linter?
NoOpTracer does not seem to be used in prod code currently but I think it's best to avoid any possible memory leak, even in UTs

Reply:
Removed the struct altogether- it wasn't used anywhere.

Reply:
The consumer should be ending the span. The linter is complaining that the returned span is unassigned so there could be a possible memory leak